### PR TITLE
configure json-facet via native ruby hashes converted to JSON

### PR DESCRIPTION
**Pull request by magibney**
This is cleaner and more elegant all-around, but the
main practical benefit is that it supports much more
robust/generic/nuanced parsing of json-facet response
objects

Squashed commit of the following:

commit 98d4434884b0e7e65fa5dbbf8b8e1cab09b769d0
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Fri Oct 2 14:15:02 2020 -0400

    X7: be more parsimonious with hierarchy components

    we don't actually need the top-level of the hierarchy

commit e499d2e0974c94bf0ea8a39753d63e6597511772
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Fri Oct 2 14:11:03 2020 -0400

    X6: some trivial cleanup

commit 3ff6a47bd2230dfc78db39b9b4b8b4cc534d6c49
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Fri Oct 2 14:10:36 2020 -0400

    X5: fix hierarchy stuff (e.g., db_category_f)

commit 4241b0387803af07790dbb59ea4cc27df3f17553
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Fri Oct 2 13:12:50 2020 -0400

    X4: subject_specialists now working too

commit 06864cc904e1a20550fcac04e9e6b3f1f73f74d3
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Fri Oct 2 11:43:59 2020 -0400

    X3: subject_specialists initial; not yet working

commit 169450c1f66f4d3e0d98534a5103d6324171e8f3
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Fri Oct 2 02:29:38 2020 -0400

    X2: initial working through to render!

commit 7e9d2a633858252b805779b1eaaabfddab56681a
Author: Michael Gibney <michael@michaelgibney.net>
Date:   Thu Oct 1 17:13:34 2020 -0400

    X1: begin make json.facet ruby objects

### CODE REVIEW

**Review by mksndz — CHANGES_REQUESTED** [review]

File: app/helpers/blacklight_helper.rb [hunk 1]
```diff
@@ -10,15 +10,21 @@ def render_search_bar
   end
 
   def render_expert_help(specialists)
-    if specialists.blank? || specialists.items.first.hits < 30_000
+    return unless specialists
+    specialists = specialists.items[0].subs.each_with_object([]) do |(k, v), arr|
+      if k != :count && v.subs[:r1]
+        arr << v
+      end
+    end.sort! { |a,b| b.subs[:r1][:relatedness] <=> a.subs[:r1][:relatedness] }
```
[the remark this answers is not in the document]
Comment (mksndz) [reply app/helpers/blacklight_helper.rb:18]: You've cleared some of this up for me (thanks :)), but I think the sorting could be refactored out into it's own method pretty simply here to make things more easily comprehensible

[the remark this answers is not in the document]
Comment (magibney) [reply app/helpers/blacklight_helper.rb]: Sounds good, thanks! Just pushed 47cd813 in response to this ...

File: app/controllers/catalog_controller.rb [hunk 2]
```diff
@@ -293,111 +289,104 @@ def expire_session
...
-        'subject_f: {',
-        'type: terms,',
-        'prefix: $top_level_term--,',
-        'field: subject_f,',
-        'limit: 5',
-        '}',
-        '}',
-        '}',
-        '}'].join
+    @@DATABASE_CATEGORY_TAXONOMY = {
```
[the remark this answers is not in the document]
Comment (magibney) [reply app/controllers/catalog_controller.rb]: Ah ok; something along the lines of 1758dd4 look ok?

**Review by mksndz — APPROVED** [review]
:+1: :zap:

**Pull request merged**